### ISSUE TITLE
Bug: DOL_MAIN_URL_ROOT not detecting HTTPS environment correctly

### ISSUE THREAD

**Issue by marcosgdf** [Dolibarr/dolibarr#6058]
# Bug
Having a Dolibarr installation under an https environment setup, DOL_MAIN_URL_ROOT still returns an `http://` uri.

```php
var_dump(DOL_MAIN_URL_ROOT) = 'http://...'
var_dump($_SERVER['HTTPS']) = 'on'
```

Problem found in htdocs/filefunc.inc.php line 223.

## Environment
- **Version**: 4.0.2 and lower

**Comment by eldy** [comment]
What is value of var_dump($_SERVER["SERVER_PORT"]) ?

**Comment by eldy** [comment]
If you have :

var_dump(DOL_MAIN_URL_ROOT) = 'http://...'
var_dump($_SERVER['HTTPS']) = 'on'

then test 
((empty($_SERVER["HTTPS"]) || $_SERVER["HTTPS"] != 'on') 
should be false

then
(((empty($_SERVER["HTTPS"]) || $_SERVER["HTTPS"] != 'on') && (empty($_SERVER["SERVER_PORT"])||$_SERVER["SERVER_PORT"]!=443))

is false

so 
$tmp='http'.(((empty($_SERVER["HTTPS"]) || $_SERVER["HTTPS"] != 'on') && (empty($_SERVER["SERVER_PORT"])||$_SERVER["SERVER_PORT"]!=443))?'':'s')

shoud return tmp='https'

Did i miss something ?

**Comment by marcosgdf** [comment]
I did a little more research and I found the problem!

There is no autodetection for a normal HTTPS environment as Dolibarr only uses the value in `$dolibarr_main_url_root` in the config file.

If not defined, Dolibarr doesn't run.

So the check for HTTPs only works in the case that Dolibarr is not installed in the root directory (if `$found` equals 0).

![captura de pantalla 2016-12-27 a las 14 48 11](https://cloud.githubusercontent.com/assets/168744/21500892/b8011aac-cc43-11e6-86b0-2547d734600f.png)

![captura de pantalla 2016-12-27 a las 14 49 37](https://cloud.githubusercontent.com/assets/168744/21500893/bb8550a8-cc43-11e6-82e1-e5dcef74cb91.png)

Is this the expected behaviour? Should `$dolibarr_main_url_root` contain the protocol?
